### PR TITLE
Support aboring in RobotCommandExecutor

### PR DESCRIPTION
**Pull request by mitsuharu-kojima**

### CODE REVIEW

File: openrr-teleop/src/control_nodes_config.rs [hunk 1]
```diff
@@ -56,20 +56,23 @@ impl ControlNodesConfig {
         ik_solver_with_chain_map: &HashMap<String, Arc<IkSolverWithChain>>,
         move_base: Option<Arc<dyn MoveBase>>,
         joints_poses: Vec<JointsPose>,
-    ) -> Vec<Box<dyn ControlNode>> {
-        let mut nodes: Vec<Box<dyn ControlNode>> = vec![];
+    ) -> Vec<Arc<Box<dyn ControlNode>>> {
```
Comment (taiki-e) [inline openrr-teleop/src/control_nodes_config.rs:59]:
```suggestion
    ) -> Vec<Arc<dyn ControlNode>> {
```

Double boxing here is probably not necessary.

Comment (mitsuharu-kojima) [reply openrr-teleop/src/control_nodes_config.rs]: https://github.com/openrr/openrr/pull/448/checks?check_run_id=3550417379#step:7:611

I couldn't solve the error...

Comment (taiki-e) [reply openrr-teleop/src/control_nodes_config.rs]: I guess that error can be solved by adding auto_impl(Arc) to ControlNode.

```diff
- #[auto_impl(Box)]
+ #[auto_impl(Box, Arc)]
  pub trait ControlNode: Send + Sync {
```

File: openrr-teleop/src/control_nodes_config.rs [hunk 2]
```diff
@@ -56,20 +56,23 @@ impl ControlNodesConfig {
...
-        let mut nodes: Vec<Box<dyn ControlNode>> = vec![];
+    ) -> Vec<Arc<dyn ControlNode>> {
+        let mut nodes: Vec<Arc<dyn ControlNode>> = vec![];
 
         for joy_joint_teleop_config in &self.joy_joint_teleop_configs {
-            nodes.push(Box::new(JoyJointTeleopNode::new_from_config(
+            nodes.push(Arc::new(Box::new(JoyJointTeleopNode::new_from_config(
                 joy_joint_teleop_config.config.clone(),
                 joint_trajectory_client_map[&joy_joint_teleop_config.client_name].clone(),
                 speaker.clone(),
```
Comment (OTL) [inline openrr-teleop/src/control_nodes_config.rs:66]: I think you can remove `Box::new()` everywhere.

File: openrr-teleop/src/control_nodes_config.rs (hunk 1 above)
Comment (mitsuharu-kojima) [reply openrr-teleop/src/control_nodes_config.rs]: https://github.com/openrr/openrr/pull/448/checks?check_run_id=3551247005#step:7:614
https://github.com/openrr/openrr/pull/448/commits/accf9f940470b90e92ef703a7e3437e45c5f2123

File: openrr-teleop/src/control_nodes_config.rs (hunk 2 above)
Comment (OTL) [reply openrr-teleop/src/control_nodes_config.rs:66]: Sorry, this comment was toooo late...

**Comment by mitsuharu-kojima** [comment]
I split pull requests.  https://github.com/openrr/openrr/pull/450

File: openrr-teleop/src/control_nodes_config.rs (hunk 1 above)
Comment (mitsuharu-kojima) [reply openrr-teleop/src/control_nodes_config.rs]: Done

**Review by taiki-e — APPROVED** [review]

**Review by OTL — APPROVED** [review]

**Pull request merged**